File: apps/demo-be-csharp-aspnetcore/tests/DemoBeCsas.Tests/Integration/Steps/HealthSteps.cs
using System.Text.Json;
using DemoBeCsas.Tests.ScenarioContext;
using FluentAssertions;
using Reqnroll;
using Xunit;

namespace DemoBeCsas.Tests.Integration.Steps;

[Binding]
[Trait("Category", "Integration")]
public class HealthSteps(ServiceLayer svc, SharedState state)
{
    [When(@"^an operations engineer sends GET \/health$")]
    public void WhenGetHealth()
    {
        state.LastResponse = svc.HealthCheck();
    }

    [When(@"^an unauthenticated engineer sends GET \/health$")]
    public void WhenUnauthenticatedGetHealth()
    {
        state.LastResponse = svc.HealthCheck();
    }

    [Then("the health status should be {string}")]
    public void ThenHealthStatus(string expectedStatus)
    {
        state.LastResponse.Should().NotBeNull();
        var body = state.LastResponse!.Body;
        using var doc = JsonDocument.Parse(body);
        doc.RootElement.GetProperty("status").GetString().Should().Be(expectedStatus);
    }

    [Then("the response should not include detailed component health information")]
    public void ThenNoDetailedHealthInfo()
    {
        state.LastResponse.Should().NotBeNull();
        var body = state.LastResponse!.Body;
        body.Should().NotContain("components");
        body.Should().NotContain("details");
    }
}

File: apps/organiclever-be/tests/OrganicLeverBe.Tests/Integration/Steps/HealthSteps.fs
module OrganicLeverBe.Tests.Integration.Steps.HealthSteps

open System.Text.Json
open TickSpec
open Xunit
open OrganicLeverBe.Tests.State
open OrganicLeverBe.Tests.Integration.Steps.CommonSteps

let private doHealthRequest (state: StepState) =
    let response =
        state.HttpClient.GetAsync("/api/v1/health")
        |> Async.AwaitTask
        |> Async.RunSynchronously

    let body =
        response.Content.ReadAsStringAsync()
        |> Async.AwaitTask
        |> Async.RunSynchronously

    let status = int response.StatusCode
    applyResult status body state

[<When>]
let ``an operations engineer sends GET /health`` (state: StepState) = doHealthRequest state

[<When>]
let ``an unauthenticated engineer sends GET /health`` (state: StepState) = doHealthRequest state

[<Then>]
let ``the health status should be "(.+)"`` (status: string) (state: StepState) =
    let doc = JsonDocument.Parse(state.ResponseBody.Value)
    let actual = doc.RootElement.GetProperty("status").GetString()
    Assert.Equal(status, actual)
    state

[<Then>]
let ``the response should not include detailed component health information`` (state: StepState) =
    let doc = JsonDocument.Parse(state.ResponseBody.Value)
    let mutable hasComponents = false

    for prop in doc.RootElement.EnumerateObject() do
        if prop.Name = "components" || prop.Name = "details" then
            hasComponents <- true

    Assert.False(hasComponents, "Response should not include component details")
    state
